### ISSUE TITLE
Make the "5 NOT FOUND" / "Exception occurred in retry method that was not classified as transient" error message specific and useful

### ISSUE THREAD

**Issue by dandv** [firebase/firebase-admin-node#2732]
I'm raising this specific issue because this "5 NOT FOUND" error is uncharacteristically (for a Google API) unhelpful, confusing and cryptic.

When it comes from using Firestore, it can mean that,
* [the collection/document/subcollection//... path sequence wasn't respected](https://stackoverflow.com/questions/75251053/firestore-exception-occurred-in-retry-method-that-was-not-classified-as-transien)
* [Firestore was not properly set up from the console](https://github.com/firebase/firebase-admin-node/issues/2563#issuecomment-2123199809)
* [the `(default)` database was deleted](https://github.com/firebase/firebase-admin-node/issues/2563#issuecomment-2227108636)
* [the custom database ID wasn't specified](https://github.com/firebase/firebase-admin-node/issues/2563#issuecomment-2228809974)
* [a cloud function was idle for a long time](https://github.com/firebase/firebase-functions/issues/536#issuecomment-514938881)

Also, its `note` field in my case was set to "Exception occurred in retry method that was not classified as transient", but the `FirebaseFirestoreError` type doesn't include a `.note` field. (`import { FirebaseFirestoreError } from 'firebase-admin/firestore';`). There was also an empty `details` field that was not defined in `FirebaseFirestoreError`.

@dconeybe any chance for this error message to received some TLC?

**Comment by dconeybe** [comment]
Assigning @wu-hui as I don't have any special knowledge of this issue, other than having looked into one of the "NOT FOUND" issues linked to.

**Comment by mrmuffyman** [comment]
Thanks a ton @dandv for bringing this up. As a new user, I was stuck for 3 hours trying to figure out this error message. Tons of confusing information on the web on how to just get a simple db write to work but turns out I just wasn't supposed to rename my database on setup.

**Comment by efshelper** [comment]
I got this same `5 NOT_FOUND` error while following the [Firestore Quickstart](https://firebase.google.com/docs/firestore/quickstart), which appears to be incorrect. it advises you to generate the private key / JSON file from the Google Cloud console.

<img width="830" alt="Image" src="https://github.com/user-attachments/assets/ed5b0248-8fb1-42df-9e10-b486acb23f37" />

this error only went away when I generated my private key / JSON file from the Firebase Console instead (not the Google Cloud console).

### 1. Firebase Console > Project settings

<img width="373" alt="Image" src="https://github.com/user-attachments/assets/05625b8b-6826-4329-8a48-6e80b45d2628" />

### 2. Project Settings > Service accounts tab

<img width="480" alt="Image" src="https://github.com/user-attachments/assets/d8c40cc9-34da-4a30-87fe-1ef8acf85680" />


### 3. Scroll Down to "Generate new private key"

<img width="265" alt="Image" src="https://github.com/user-attachments/assets/82153a78-6972-4b23-bb32-6085e62c671c" />

Now use that JSON file to initialize the SDK.

**Comment by dails08** [comment]
Yes please. I've followed and refollowed the examples and cannot get a simple write (or read) to work, and this opaque error message is maddening. I've tried all the fixes people have mentioned.

**Comment by lahirumaramba** [comment]
As a workaround, could you fix the issue by setting the databaseId?

```js
getFirestore(app: FirebaseApp, databaseId: string);
```

(or see: https://github.com/firebase/firebase-admin-node/issues/2563#issuecomment-2994418683)
